### PR TITLE
[MINOR][SDK-1840] add client hints to replace useragent for supported chromium browsers

### PR DESCRIPTION
**Pull request by JagadeeshKaricherla-branch**
# Pull Request Template

## Description
Chromium is going to deprecate full User Agents and will start masking/reducing/freeze the user agent. The alternative for this is to use the client hints.
This PR :
- adds method to retrieve high entropy useragent data like model, platform version
- modify API request params to include useragent data model, os version
- add tests

Fixes # (issue) : SDK-1840

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit test
- [x] Integration test

## JS Budget Check

Please mention the size in kb before abd after this PR

| Files            | Before      | After       | 
| -----------      | ----------- | ----------- |
| dist/build.js.   |     161 kb        |      162 kb       |
| dist/build.min.js|      75 kb       |        75 kb     |

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Mentions: 
List the person or team responsible for reviewing proposed changes.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

File: src/journeys_utils.js [hunk 1]
```diff
@@ -857,14 +857,18 @@ journeys_utils._handleJourneyDismiss = function(eventName, storage, banner, temp
...
 journeys_utils._getPageviewMetadata = function(options, additionalMetadata) {
-	return utils.merge({
+	var pageviewMetadata = utils.merge({
 		"url": options && options.url || utils.getWindowLocation(),
 		"user_agent": navigator.userAgent,
 		"language": navigator.language,
 		"screen_width": screen.width || -1,
 		"screen_height": screen.height || -1,
-		"window_device_pixel_ratio": window.devicePixelRatio || 1
+		"window_device_pixel_ratio": window.devicePixelRatio || 1,
```
Comment (gdeluna-branch) [inline src/journeys_utils.js:866]: Is this default safe/why not -1 if not found?

File: src/1_utils.js [hunk 2]
```diff
@@ -1268,3 +1271,35 @@ utils.delay = function(operation, delay) {
 
 	setTimeout(operation, delay);
 };
+
+/**
+ * gets client hints for supportd browser.
```
Comment (gdeluna-branch) [inline src/1_utils.js:1276]: nit: typo `supported`

**Comment by gdeluna-branch** [comment]
Changes look good, only concern is default value for pixel ratio

**Review by gdeluna-branch — APPROVED** [review]

File: dist/build.js [hunk 3]
```diff
@@ -1626,6 +1629,15 @@ utils.getBooleanOrNull = function(a) {
 utils.delay = function(a, b) {
   isNaN(b) || 0 >= b ? a() : setTimeout(a, b);
 };
+utils.getClientHints = function() {
```
Comment (ahmednawar) [inline dist/build.js:1632]: can you add a comment on why we are doing this for reference?

Comment (JagadeeshKaricherla-branch) [reply dist/build.js:1632]: build.js is a generated file by google closure compiler which removes all the comments.
The original code file has a comment about the function
https://github.com/BranchMetrics/web-branch-deep-linking-attribution/blob/96550a89fff6e93f1fdbb53edf8e6f3a4796e299/src/1_utils.js#L1276

File: src/journeys_utils.js (hunk 1 above)
Comment (JagadeeshKaricherla-branch) [reply src/journeys_utils.js:866]: for older browsers that don't support devicePixelRatio, we fallback to 1 which is classic or standard display with zoom at 100%. You rarely see any devices with device pixel ratio in negative, https://www.mydevice.io/#tab1
This is pretty old code and not touched as part of this PR  and we can evaluate based on our browser support, if we need these fallbacks to continue.

**Review by ahmednawar — APPROVED** [review]

File: dist/build.js [hunk 4]
```diff
@@ -1626,6 +1629,15 @@ utils.getBooleanOrNull = function(a) {
...
+  navigator.userAgentData ? navigator.userAgentData.getHighEntropyValues(["model", "platformVersion"]).then(function(a) {
```
Comment (mack-branch) [inline dist/build.js:1633]: does this api expose brand on mobile devices? can we send that too?

Comment (JagadeeshKaricherla-branch) [reply dist/build.js:1633]: @mack-branch  brand is a low entropy value which can be directly accessed using
`console.log(navigator.userAgentData.brands);`
which results in an array something like 

```
[
    {
        "brand": "Chromium",
        "version": "110"
    },
    {
        "brand": "Not A(Brand",
        "version": "24"
    },
    {
        "brand": "Google Chrome",
        "version": "110"
    }
]
```
To check whether the device type is a mobile or not. we can use `console.log(navigator.userAgentData.mobile);` :  true if this is a mobile device.

**Pull request merged**